src/libYARP_sig/include/yarp/sig/PointCloud.h

```diff
@@ -0,0 +1,597 @@
+         */
+        virtual size_t height() const;
+
+        /**
+         * @brief
+         * @return the width specified int the yarp::sig::PointCloudNetworkHeader.
+         */
+        virtual size_t width() const;
+
+        /**
+         * @brief
+         * @return the enum representing the point type.
+         */
+        virtual int getPointType() const;
+
+        virtual yarp::os::Type getType() override;
+
+        /**
+         * @brief
+         * @return true if the point cloud is organized in an image-like structure
+         */
+        virtual bool isOrganized();
+
+        /**
+         * @brief isDense
+         * @return true if the point cloud doesn't contain NaN or Inf values
+         */
+        virtual inline bool isDense() const
+        {
+            return header.isDense;
+        }
+
+    protected:
+
+        PointCloudBase();
+
+        virtual void copyFromRawData(const char* dst, const char* source, std::vector<int> &recipe);
+
+        virtual std::vector<int> getComposition(int type_composite) const;
+
+        virtual size_t pointType2Size(int type) const;
+
+        virtual size_t getOffset(int type_composite, int type_basic) const;
+
+        yarp::sig::PointCloudNetworkHeader    header;
+
+    };
+
+    template <class T>
+    /**
+     * @brief The PointCloud class.
+     */
+    class PointCloud : public PointCloudBase
+    {
+        static_assert(std::is_same<T, XY_DATA>::value ||
+                      std::is_same<T, XYZ_DATA>::value ||
+                      std::is_same<T, NORMAL_DATA>::value ||
+                      std::is_same<T, XYZ_RGBA_DATA>::value ||
+                      std::is_same<T, XYZ_I_DATA>::value ||
+                      std::is_same<T, INTEREST_POINT_XYZ_DATA>::value ||
+                      std::is_same<T, XYZ_NORMAL_DATA>::value ||
+                      std::is_same<T, XYZ_NORMAL_RGBA_DATA>::value, "yarp::sig::PointCloud<T>: T chosen is not supported");
+    public:
+
+        /**
+         * @brief PointCloud, default constructor.
+         */
+        PointCloud()
+        {
+            data.clear();
+            setPointType();
+        }
+
+        /**
+         * @brief PointCloud, copy constructor.
+         * Clones the content of another point cloud.
+         * @param alt the point cloud to clone.
+         */
+        template<class T1>
+        PointCloud(const PointCloud<T1>& alt)
+        {
+            setPointType();
+            copy<T1>(alt);
+            return;
+
+        }
+
+        /**
+         * @brief Resize the PointCloud.
+         * @param width.
+         * @param height.
+         */
+        virtual void resize(size_t width, size_t height)
+        {
+            header.width = width;
+            header.height = height;
+            data.resize(width * height);
+        }
+
+        /**
+         * @brief Resize the PointCloud.
+         * @param width.
+         * Calling this function the cloud will become NOT organizedS
```

Comment:
I am not sure what you mean by "will become NOT organizedS"

Reply:
A point cloud is organized if `height > 1`. If you resize it with only one argument it will become not organized.
An organized point cloud is treatable like an image, accessing the pixels with u and v.

Reply:
My suggestion here is to rewrite the sentence as follows:
"This function resizes the point cloud and has the side effect of reducing the height to 1. Thus, by definition, the point cloud becomes unorganized."

Reply:
Even better with `@note`